Simple stuff, setting up python and installing dep

FizzWizZleDazzle opened the pull request:
Please review my bash code

FizzWizZleDazzle committed:
Simple stuff, setting up python and installing dep

brunoarpa committed:
yes neuron.py

FizzWizZleDazzle committed:
Fix neuron structure

x is not part of neuron

FizzWizZleDazzle committed:
Backwards function for neuron

I don't think we should hardcode sigmoid activation function. Please advise.

FizzWizZleDazzle commented:
Waiting for pull request #6

brunoarpa committed:
relu neuron.py

brunoarpa committed:
Merge pull request #6 from TrainPioneers/branch-prealpha-patch-1

Backwards function for neuron

FizzWizZleDazzle committed:
Activation was declared but never used. Fixed

scarlettmparker commented on scripts/install_dependencies.sh:
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+pip install ../../backend/sentiment/requirements.txt
```
this is wrong, it should be

pip install -r ../backend/requirements.txt

-r is needed anyway for requirements files but you also put the wrong directory here.

FizzWizZleDazzle replied:
Thanks for reminding

FizzWizZleDazzle replied:
I was really asking if #!/usr/bin/env bash is correct

scarlettmparker commented on backend/sentiment/prototype/neuron.py:
```diff
@@ -0,0 +1,45 @@
+    w: np.array  # weights
+    b: float = 0.0  # bias
+    learning_rate: float = 0.01  # learning rate for gradient descent
+    activation = None  # activation function, float
```
though this hasn't been decided on yet it should still be set up with default functions, perhaps set a default if self.activation = none

FizzWizZleDazzle replied:
It is

FizzWizZleDazzle replied:
Sure. relu?

scarlettmparker replied:
forgot to say yes to this but sure i see it has already been done tho

scarlettmparker commented on backend/sentiment/prototype/neuron.py:
```diff
@@ -0,0 +1,45 @@
+        self.w -= self.learning_rate * d_z * x  # gradient descent step for weights
+        self.b -= self.learning_rate * d_z      # gradient descent step for bias
+
+    def sigmoid(self, z: float) -> float:
```
should these not be static methods?

FizzWizZleDazzle replied:
That's what I was talking about. It shouldn't be hardcoded into the neuron dataclass.

scarlettmparker reviewed (commented):
install dependencies script is wrong + could change a few things in neuron.py

FizzWizZleDazzle committed:
Fix install_dependencies.sh

FizzWizZleDazzle committed:
Fix neuron.py

Make activation functions static and make relu the default.

FizzWizZleDazzle commented:
Waiting on #7 . Fixes the problems.

brunoarpa committed:
Merge pull request #7 from TrainPioneers/branch-prealpha-patch-2

ye mb outside better neuron.py

scarlettmparker committed:
re-arranged backend files + set up front-end files

feel free to move the scripts folder back but i think it should stay separate for front-end and backend and not be in the parent directory. it was still broken so i fixed it.

set up a venv so any new packages can be frozen into requirements easily. will start a new branch for front-end stuff after this

FizzWizZleDazzle commented:
Why we don't need pyc nor 1000000 lines

FizzWizZleDazzle committed:
Delete backend/venv directory

scarlettmparker commented:
whoops i should've put that in the gitignore my bad idk what was going through my head when i wrote that commit

scarlettmparker reviewed (approved):
i say we merge and split off into branches for backend frontend im pretty sure the usr/bin/env bash thing works

FizzWizZleDazzle commented:
sure
